CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/GiniNetworkingScreenAPICoordinator.swift

```diff
@@ -134,12 +134,14 @@ import GiniBankAPILibrary
 extension GiniNetworkingScreenAPICoordinator {
     fileprivate func startAnalysis(networkDelegate: GiniCaptureNetworkDelegate) {
         self.documentService.startAnalysis { result in
-            switch result {
-            case .success(let extractions):
-                self.deliver(result: extractions, and: self.documentService.document, to: networkDelegate)
-            case .failure(let error):
-                guard error != .requestCancelled else { return }
-                networkDelegate.displayError(errorType: ErrorType(error: error), animated: true)
+            DispatchQueue.main.async {
```

Comment:
It seems that also `self.deliver(result: extractions, and: self.documentService.document, to: networkDelegate)` is switching to the main thread and this can potentially lead to recursive or nested asynchronous calls. This can cause unintended consequences, such as excessive resource consumption or unexpected order of execution. 
Having two `DispatchQueue.main.async` calls does not inherently cause a problem. However, it's important to consider the order and timing of the code execution to avoid unexpected behavior.